test/metabase/models/setting_test.clj

```diff
@@ -1190,3 +1191,60 @@
                             :previous-value nil
                             :new-value      "AUDIT ME"}}
                  (mt/latest-audit-log-entry :setting-update))))))))
+
+(deftest realize-throwing-test
+  (testing "The realize function ensures all nested lazy values are calculated"
+    (let [ok (lazy-seq (cons 1 (lazy-seq (list 2))))
+          ok-deep (lazy-seq (cons 1 (lazy-seq (list (lazy-seq (list 2))))))
+          shallow (lazy-seq (cons 1 (throw (ex-info "Surprise!" {}))))
+          deep (lazy-seq (cons 1 (cons 2 (list (lazy-seq (throw (ex-info "Surprise!" {})))))))]
+      (is (= '(1 2) (#'setting/realize ok)))
+      (is (= '(1 (2)) (#'setting/realize ok-deep)))
+      (doseq [x [shallow deep]]
+        (is (thrown-with-msg?
+              clojure.lang.ExceptionInfo
+              #"^Surprise!$"
+              (#'setting/realize x)))))))
+
+(deftest valid-json-setting-test
+  (mt/with-temp-env-var-value ["MB_TEST_JSON_SETTING" "[1, 2]"]
+    (is (nil? (setting/validate-settings-formatting!)))))
+
+(defn- get-parse-exception [raw-value]
+  (mt/with-temp-env-var-value ["MB_TEST_JSON_SETTING" raw-value]
+    (try
+      (setting/validate-settings-formatting!)
+      (throw (java.lang.RuntimeException. "This code should never be reached."))
+      (catch java.lang.Exception e e))))
+
+(defn- assert-parser-exception! [ex cause-message]
+  (is (= "Invalid JSON configuration for setting: test-json-setting" (ex-message ex)))
+  (is (= cause-message (ex-message (ex-cause ex)))))
+
+(deftest invalid-json-setting-test
```

Comment:
Last thing: since you've gone to the trouble of implementing it, we should test for other types of settings that can also have invalid values, e.g. csv, integer, and boolean

Reply:
Discussed offline, and leaving for a further PR